FreeRTOS/Test/CMock/tasks/Makefile

```diff
@@ -0,0 +1,215 @@
+	    sed -i "1 i\#define $$guard" $(@D)/$$(basename $$h_file);          \
+	    sed -i "1 i\#ifndef $$guard" $(@D)/$$(basename $$h_file);          \
+	    sed -i -e '$$a#endif' $(@D)/$$(basename $$h_file);                 \
+        done
+
+.SECONDARY:
+# Build ALL the mock objects for the config
+# This target is called ONLY once per configuration pattern (%)
+$(MOCK_OBJS) : $(MOCK_SRC_LIST)
+	$(eval INCLUDE = -I$(SCRATCH_DIR)/$*/include)
+	$(eval INCLUDE += -I$(SCRATCH_DIR)/$*/mocks)                           \
+	$(eval INCLUDE += -I$(KERNEL_DIR)/include)                             \
+	for files in $^ ; do                                                   \
+       	    new_name=$${files%.c}.o;                                           \
+	    $(CC) -c $$files $(INCLUDE) $(CFLAGS) -fPIC -DUNITY_EXCLUDE_MATH_H \
+		-DUNITY_EXCLUDE_STDINT_H -DUNITY_OUTPUT_CHAR      -Wno-error   \
+		-o $$new_name ;                                                \
+        done
+#find $(@D)  -type f ! -name 'mock_*' -delete
+
+.SECONDARY:
+# Build test_runner.o from test_runner.c
+$(SCRATCH_DIR)/%/test_runner.o : $(SCRATCH_DIR)/%/test_runner.c                \
+                                 $(SCRATCH_DIR)/%/$(PROJECT).o
+	$(eval INCLUDE = -I$(SCRATCH_DIR)/$*/include)
+	$(eval INCLUDE += -I$(SCRATCH_DIR)/$*/mocks)
+	$(eval INCLUDE += -I$(KERNEL_DIR)/include)
+	$(CC) -c $< $(INCLUDE) $(CPPFLAGS)  $(CFLAGS) -Wno-error -o $@
+
+.SECONDARY:
+# Build tasks_utest.o from tasks_utest_%.c
+$(SCRATCH_DIR)/%/$(PROJECT)_utest.o : $(PROJECT_DIR)/$(PROJECT)_utest_%.c      \
+                                      | directories
+	$(eval INCLUDE = -I$(SCRATCH_DIR)/$*/include)
+	$(eval INCLUDE += -I$(SCRATCH_DIR)/$*/mocks)
+	$(eval INCLUDE += -I$(KERNEL_DIR)/include)
+	$(CC) -c $< $(INCLUDE) $(CPPFLAGS)  $(CFLAGS) -Wno-error -o $@
+
+.SECONDARY:
+# Build tasks.so from tasks.c with the custom FreeRTOSConfig.h and place it in
+# its configuration directory
+$(SCRATCH_DIR)/%/$(PROJECT).o : $(KERNEL_DIR)/$(PROJECT).c | directories
+	$(eval INCLUDE = -I$(SCRATCH_DIR)/$*/include)
+	$(eval INCLUDE += -I$(SCRATCH_DIR)/$*/mocks)                           \
+	$(eval INCLUDE += -I$(KERNEL_DIR)/include)                             \
+	$(CC) -c $< $(INCLUDE) $(CPPFLAGS)                                     \
+	      $(CFLAGS) -include list_macros.h $(COVERAGE_OPTS) -Wno-error -o $@
+
+
+.SECONDARY:
+# Generate test_runner.c which contains the main function from the test file
+$(SCRATCH_DIR)/%/test_runner.c : $(PROJECT_DIR)/$(PROJECT)_utest_%.c           \
+                                 Makefile | directories
+	ruby $(UNITY_BIN_DIR)/generate_test_runner.rb $(EXEC)_$*.c             \
+                       $(PROJECT_DIR)/$(PROJECT).yml $@
+
+# Create needed directories
+directories :
+	-mkdir   -p  $(SCRATCH_DIR)
+	-mkdir   -p  $(CONFIG_DIRS)
+	-mkdir   -p  $(MOCK_DIRS)
+	-mkdir   -p  $(CPP_DIRS)
+	-mkdir   -p  $(INCLUDE_DIRS)
+
+.PRECIOUS: build/%.o
```

Comment:
does the build/*.o path exist?

Reply:
it is a % so it catches anything with any path